processor/eventfilter/eventfilter.go

```diff
@@ -69,3 +77,325 @@ func GetSupportedMessageEvents(destination *backendconfig.DestinationT) ([]strin
+		// Only when 1 & 2 are true, we would allow the event to flow through to server
+		// As when this will be called, we would have already checked if event.type in supportedMessageTypes
+		DefaultBehaviour: transformerEvent.Destination.IsProcessorEnabled && isSupportedMsgType,
+	})
+
+	if !allow {
+		return allow, failedResponse
+	}
+	// srcType.connectionMode.[eventProperty] filtering -- ENDS
+
+	return true, nil
+}
+
+/*
+Lets the caller know if the event is allowed to flow through server for a `specific destination`
+Introduced to support hybrid-mode, cloud-mode in a more scalable way
+
+The template inside `destinationDefinition.Config.supportedConnectionModes` would look like this
+```
+
+	[sourceType]: {
+		[connectionMode]: {
+			[eventProperty]: [...supportedEventPropertyValues]
+		}
+	}
+
+```
+
+Example:
+
+	{
+		...
+		"supportedConnectionModes": {
+			"unity": {
+				"cloud": {
+					"messageType": { "allowAll": true },
+				},
+				"hybrid": {
+					"messageType": { "allowedValues": ["group","track"] },
+				}
+			},
+			"web": {
+				"cloud": {
+					"messageType": { "allowedValues": ["group","track"] },
+				},
+				"device": {
+					"messageType": { "allowAll": true },
+				}
+			}
+		},
+		...
+	}
+*/
+func FilterUsingSupportedConnectionModes(connectionModeFilterParams ConnectionModeFilterParams) (bool, *transformer.TransformerResponseT) {
+	destination := connectionModeFilterParams.Destination
+	srcType := connectionModeFilterParams.SrcType
+	messageType := connectionModeFilterParams.Event.MessageType
+	evaluatedDefaultBehaviour := connectionModeFilterParams.DefaultBehaviour
+
+	/*
+		Cases:
+		1. if supportedConnectionModes is not present -- rely on default behaviour
+		2. if sourceType not in supportedConnectionModes && sourceType is in supportedSourceTypes (the sourceType is not defined) -- rely on default behaviour
+		3. if sourceType = {} -- don't send the event(silently drop the event)
+		4. if connectionMode not in supportedConnectionModes.sourceType (some connectionMode is not defined) -- don't send the event(silently drop the event)
+		5. if sourceType.connectionMode = {} -- rely on default behaviour
+		6. "device" is defined as a key for a sourceType -- don't send the event
+		7. if eventProperty not in sourceType.connectionMode -- rely on property based default behaviour
+		8. if sourceType.connectionMode.eventProperty = {} (both allowAll, allowedValues are not included in defConfig) -- rely on default behaviour
+		9. if sourceType.connectionMode.eventProperty = { allowAll: true, allowedValues: ["track", "page"]} (both allowAll, allowedValues are included in defConfig) -- rely on default behaviour
+	*/
+
+	destConnModeI := misc.MapLookup(destination.Config, "connectionMode")
+	if destConnModeI == nil {
+		pkgLogger.Debug("connectionMode not present, filtering event based on default behaviour")
+		return evaluatedDefaultBehaviour, nil
+	}
```

Comment:
Instead of ignoring the error, we can use for better log message, plus is more idiomatic to check for absent error rather than nil

```suggestion
	destConnModeI, err := misc.MapLookup(destination.Config, "connectionMode")
	if err != nil {
		pkgLogger.Debug("connectionMode not present, filtering event based on default behaviour: %v", err)
		return evaluatedDefaultBehaviour, nil
	}
```

Reply:
`MapLookup` returns an `interface{}` only. Hence the check condition was written as such.